tools/fetch_fasta_from_ncbi/retrieve_fasta_from_NCBI.py

```diff
@@ -268,10 +270,19 @@ def __main__():
         kwargs['filename'] = options.logfile
     logging.basicConfig(**kwargs)
     logger = logging.getLogger('data_from_NCBI')
-    
-    E = Eutils(options, logger)
-    E.retrieve()
 
+    query=options.query_string
+    try:
+  	bins=map(int,options.bins.split(" "))
+	for i in range(len(bins)-1):
+        	options.query_string=query+" AND "+str(bins[i]+1)+":"+str(bins[i+1])+"[Sequence Length]"
+        	print options.query_string
+        	E = Eutils(options, logger)
+        	E.retrieve()
+    except:
```

Comment:
Do not use bare Exception when possible. Usually you can test for the specific Exception like so
(KeyError would be the exception for a non-existing key in a dictionary)
```
except KeyError:
```
You can see the correct key in the traceback.